internal/jet/func_expression.go

```diff
@@ -468,6 +468,33 @@ func REGEXP_LIKE(stringExp StringExpression, pattern StringExpression, matchType
 	return newBoolFunc("REGEXP_LIKE", stringExp, pattern)
 }
 
+//----------Range Type Functions ----------------------//
+
+// LOWER_BOUND returns range expressions lower bound
+func LOWER_BOUND[T Expression](rangeExpression Range[T]) T {
+	return rangeTypeCaster[T](rangeExpression, NewFunc("LOWER", []Expression{rangeExpression}, nil))
+}
+
+// UPPER_BOUND returns range expressions upper bound
+func UPPER_BOUND[T Expression](rangeExpression Range[T]) T {
+	return rangeTypeCaster[T](rangeExpression, NewFunc("UPPER", []Expression{rangeExpression}, nil))
+}
+
+func rangeTypeCaster[T Expression](rangeExpression Range[T], exp Expression) T {
+	var i Expression
+	switch rangeExpression.(type) {
```

Comment:
I don't think this will compile. Go does not allow using generics and type switch in the same function.
I also don't see how we can return `T` in this case, so I think we should just return `Expression`. The user will have to, if needed, add type manually , like: `IntExp(UPPER_BOUND(table.Range)).ADD()`

Reply:
As you can see I used switch on first parameter not `T` itself and also here `internal/jet/func_expression_test.go:206` just tested with a simple IntegerExpression and there isn't any problem.
Returning an expression make it more simple and less type safe, but I think as you can see we can return T itself with that switch case so why not?

Reply:
Hmm. If it works it is great, I wasn't aware it was possible.